### ISSUE TITLE
Clock

### ISSUE THREAD

**Issue by jrpie** [jrpie/Launcher#12]
I got serveral requests to rework the date / clock:
* different clock faces (e.g. stacking hours and minutes)
* (add option to) follow system locale; the approach of manually adding formatting options doesn't scale well. Maybe let the user provide a formatting string? 
* disable it entirely

**Comment by toby-bro** [comment]
Didn't you already do this partially in https://github.com/jrpie/Launcher/commit/e86ed34fe5b27b74aada55ffe71b3fc7ca75a5c8 ( for the locale part ) ?

**Comment by jrpie** [comment]
Yes, should have mentioned that here ^^

**Comment by wassupluke** [comment]
Idea: add an option to apply the pretty new Dynamic color theme color to the clock? The dynamic colors look dang nice in the app settings now :)

**Comment by jrpie** [comment]
Should I add
(a) an option "use color from color theme" (i.e. only use dynamic  when the dynamic theme is used) or
(b) "use dynamic color" (independently of color theme)?

**Comment by wassupluke** [comment]
Maybe b) use dynamic color. Perhaps as a toggle inside the new RGBA sliders panel?

**Comment by jrpie** [comment]
I agree that a toggle inside the dialog would be the best UI. On the other hand I'd like to keep the color picker as generic and reusable as possible.

**Comment by wassupluke** [comment]
If there's a way to get a

**Comment by jrpie** [comment]
> Regarding "adjustable font size for the clock", Are there any preferences in how it is implemented? Default font-size? menu or slider? "Medium" vs "16"? Do we want only a few choices or many?

I think a slider with sensible minimum and maximum value should be sufficient. The size of the second text could just be hardcoded to 60% of the main text. I would not over-engineer it (e.g. a preview is of course nice, but also quite a lot of work and quite a lot of additional code to maintain).

The long-term idea is to implement #44, letting users pick whatever clock widget they like. (That's the main reason why I have not been working on this issue). I have been thinking about writing a companion app (say µClock) which could provide a selection of clocks as widgets (many people have been asking for an analog clock for example).

That's why I don't really want too much clock logic in µLauncher itself. I think the default clock should work reasonably well, but µLauncher is not a clock app, more sophisticated clocks should be provided by other apps a widgets.
An API (as discussed in #88 ) would make it possible for those widgets to interact with µLauncher functionality.
